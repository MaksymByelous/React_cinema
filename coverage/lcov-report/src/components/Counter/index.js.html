<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Counter/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/Counter</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/8</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React <span class="cstat-no" title="statement not covered" >from 'react';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ></span></span></span></span>
import Styles <span class="cstat-no" title="statement not covered" >from './styles.scss';<span class="cstat-no" title="statement not covered" ></span></span>
import { number }<span class="cstat-no" title="statement not covered" > from 'prop-types';<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span>
&nbsp;
const Counter = <span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >({</span> count }) =&gt; (</span>
<span class="cstat-no" title="statement not covered" >    &lt;section className = { Styles.counter }&gt; Posts count: {count}&lt;/section&gt;</span>
);
&nbsp;
<span class="cstat-no" title="statement not covered" >Counter.propTypes = {</span>
    count: number.isRequired
};
&nbsp;
<span class="cstat-no" title="statement not covered" >Counter.defaultProps = {</span>
    count: 0<span class="cstat-no" title="statement not covered" ></span>
};
&nbsp;
export default Counter;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Oct 29 2017 02:28:57 GMT+0300 (EEST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
