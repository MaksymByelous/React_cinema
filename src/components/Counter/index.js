import React from 'react';
import Styles from './styles.scss';
import { number } from 'prop-types';

const Counter = ({ count }) => (
    <section className = { Styles.counter }> Posts count: {count}</section>
);

Counter.propTypes = {
    count: number.isRequired
};

Counter.defaultProps = {
    count: 0
};

export default Counter;
