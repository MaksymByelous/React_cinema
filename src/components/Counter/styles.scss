@import '../../theme/palette';

.counter {
   position: relative;
   color: $paletteColor7;
   margin: 10px auto;
   padding-left: 35px;

   &::after {
       content: '';
       position: absolute;
       top: -6px;
       left: 0;
       width: 30px;
       height: 30px;
       background-image: url('../../theme/assets/post.png');
       background-repeat: no-repeat;
       background-size: 30px;
   }
}