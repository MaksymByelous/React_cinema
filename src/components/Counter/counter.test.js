import React from 'react';
import dom from 'react-test-renderer';
import Counter from './';

const renderTree = dom.create(
    <Counter count = { 3 } />
).toJSON();

test(
    'counter should correspond to snapshot', () => {
        expect(renderTree).toMatchSnapshot();
    }
);
