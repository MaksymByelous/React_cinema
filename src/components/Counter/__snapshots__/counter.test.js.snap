// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter should correspond to snapshot 1`] = `
<section
  className={undefined}
>
   Posts count: 
  3
</section>
`;
